### PR TITLE
Fix special casing for PI in model value

### LINKED ISSUE #8183: Fatal failure at /root/cvc5/src/theory/theory_model_builder.cpp:535
Hi,
For this instance, with the option `-i`,cvc5 https://github.com/cvc5/cvc5/commit/f051883afa31d9dbe14a5d1ae1c9700cffded3b1 debug build
```
$ cvc5 delta.smt2 -i
Model values in the same equivalence class 0 real.pi
Fatal failure within bool cvc5::theory::TheoryEngineModelBuilder::buildModel(cvc5::theory::TheoryModel*) at /root/cvc5/src/theory/theory_model_builder.cpp:535
Check failure

 false
Base model values in the same equivalence class 0 real.pi

Aborted (core dumped)
$ cat delta.smt2
(set-logic ALL)
(declare-fun v () Real)
(declare-fun a () (Array Real Real))
(assert (exists ((V Real)) (or (and (= 0 (/ 0 v)) (< 0 (/ 0 v))) (= 1.0 (select (store a (/ (+ 0.0 real.pi) 0.0) 0.0) 0.0)))))
(check-sat)
```

### PR DESCRIPTION
**Pull request by ajreynol**
PI should be considered a model value but not a "base" model value. This avoid spurious debug assertion failures, which are now treated as warnings.  This makes a difference when another theory e.g. UF says that PI is currently equal to a constant value `c`. Since our lemma schemas are model-based, we may require building a spurious model where PI is c, and refine afterwards.

This also fixes a bug in our extended function rewriting which would mistakenly think that PI was reduced, and hence skip a full effort check if PI was the *only* extended function in the current context.  This was previously causing a final model to be one where PI = 0.0.

Fixes #8183.

### CODE REVIEW

File: src/theory/arith/nl/ext_theory_callback.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ bool NlExtTheoryCallback::isExtfReduced(
       }
     }
   }
-  return true;
+  return false;
```
Comment (nafur) [inline src/theory/arith/nl/ext_theory_callback.cpp:132]: Which cases do change here? It is only PI?

Comment (ajreynol) [reply src/theory/arith/nl/ext_theory_callback.cpp:132]: Potentially many, the method was blatantly wrong before.  However, we don't really use this value much, it determines what is put into this temporary vector https://github.com/cvc5/cvc5/blob/master/src/theory/arith/nl/nonlinear_extension.cpp#L250.

Comment (nafur) [reply src/theory/arith/nl/ext_theory_callback.cpp:132]: So it does not only change pi?

Comment (ajreynol) [reply src/theory/arith/nl/ext_theory_callback.cpp:132]: Right, the fix also ensures that other transcendental functions are added to `nred` on that line.

**Review by nafur — APPROVED** [review]
LGTM.

**Pull request merged**